moto/resourcegroupstaggingapi/models.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import unicode_literals
+import uuid
+import boto3
+import six
+from moto.core import BaseBackend
+from moto.core.exceptions import RESTError
+
+from moto.s3 import s3_backends
+
+# Left: EC2 ElastiCache RDS ELB CloudFront WorkSpaces Lambda EMR Glacier Kinesis Redshift Route53
+# StorageGateway DynamoDB MachineLearning ACM DirectConnect DirectoryService CloudHSM
+# Inspector Elasticsearch
+
+
+class ResourceGroupsTaggingAPIBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(ResourceGroupsTaggingAPIBackend, self).__init__()
+        self.region_name = region_name
+
+        self._pages = {}
+        # Like 'someuuid': {'gen': <generator>, 'misc': None}
+        # Misc is there for peeking from a generator and it cant
+        # fit in the current request. As we only store generators
+        # theres not really any point to clean up
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    @property
+    def s3_backend(self):
+        """
+        :rtype: moto.s3.models.S3Backend
+        """
+        return s3_backends['global']
+
+    def _get_resources_generator(self, tag_filters=None, resource_type_filters=None):
+        # TODO move these to their respective backends
+
+        # Do S3
+        for bucket in self.s3_backend.buckets.values():
+            tags = []
+            for tag in bucket.tags.tag_set.tags:
+                tags.append({'Key': tag.key, 'Value': tag.value})
+
+            yield {'ResourceARN': 'arn:aws:s3:::' + bucket.name, 'Tags': tags}
+
+        # TODO add more
+
+    def get_resources(self, pagination_token=None,
+                      resources_per_page=50, tags_per_page=100,
+                      tag_filters=None, resource_type_filters=None):
+        # Simple range checning
+        if 100 >= tags_per_page >= 500:
+            raise RESTError('InvalidParameterException', 'TagsPerPage must be between 100 and 500')
+        if 1 >= resources_per_page >= 50:
+            raise RESTError('InvalidParameterException', 'ResourcesPerPage must be between 100 and 500')
```

Comment:
Should this be `if 100 >= resources_per_page >= 500:` ?

Reply:
Good spot, yeah the exception was copy/paste'd :wink:, will fix later.

Reply:
- [ ] update resources_per_page range (making a comment just to help you keep track)